ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs

```diff
@@ -0,0 +1,446 @@
+                   LookupReqs (..), RelayAccessPoint (..), PortNumber)
+import           Ouroboros.Network.PeerSelection.Types (PeerAdvertise (..))
+import           Ouroboros.Network.Server.RateLimiting
+                   (AcceptedConnectionsLimit (..))
+import           Ouroboros.Network.Snocket (TestAddress (..))
+
+import           Ouroboros.Network.Testing.ConcreteBlock (Block)
+import           Ouroboros.Network.Testing.Data.Script (Script (..))
+import           Ouroboros.Network.Testing.Utils (genDelayWithPrecision)
+import           Simulation.Network.Snocket
+                   (withSnocket,
+                   BearerInfo (..))
+
+import           Test.Ouroboros.Network.Diffusion.Node.NodeKernel (NtNAddr, randomBlockGenerationArgs, BlockGeneratorArgs)
+import           Test.Ouroboros.Network.PeerSelection.RootPeersDNS (DNSTimeout,
+                   DNSLookupDelay)
+import qualified Test.Ouroboros.Network.Diffusion.Node.NodeKernel    as Node
+import qualified Test.Ouroboros.Network.Diffusion.Node.MiniProtocols as Node
+import qualified Test.Ouroboros.Network.Diffusion.Node               as Node
+
+import           Test.QuickCheck
+                   (Arbitrary(arbitrary),
+                     Gen,
+                     choose,
+                     chooseInt,
+                     sized,
+                     vectorOf,
+                     suchThat,
+                     oneof,
+                     frequency,
+                     sublistOf
+                   )
+
+
+-- | Diffusion Simulator Arguments
+--
+-- Contains all necessary randomly generated values needed to run diffusion in
+-- simulation.
+--
+data SimArgs =
+  SimArgs
+    { saSlot :: DiffTime
+      -- ^ 'randomBlockGenerationArgs' slot duration argument
+    , saSeed :: StdGen
+      -- ^ 'randomBlockGenerationArgs' seed argument
+    , saQuota :: Int
+      -- ^ 'randomBlockGenerationArgs' quota value
+    , saMbTime :: Maybe DiffTime
+      -- ^ 'LimitsAndTimeouts' argument
+    , saRelays :: [RelayAccessPoint]
+      -- ^ 'Interfaces' relays auxiliary value
+    , saRng :: StdGen
+      -- ^ 'Interfaces' 'iRng' value
+    , saDomainMap :: Map Domain [IP]
+      -- ^ 'Interfaces' 'iDomainMap' value
+    , saAddr :: NtNAddr
+      -- ^ 'Arguments' 'aIPAddress' value
+    , saLocalRootPeers :: [(Int, Map RelayAccessPoint PeerAdvertise)]
+      -- ^ 'Arguments' 'LocalRootPeers' values
+    , saLocalSelectionTargets :: PeerSelectionTargets
+      -- ^ 'Arguments' 'aLocalSelectionTargets' value
+    , saDNSTimeoutSctipt :: Script DNSTimeout
+      -- ^ 'Arguments' 'aDNSTimeoutScript' value
+    , saDNSLookupDelayScript :: Script DNSLookupDelay
+      -- ^ 'Arguments' 'aDNSLookupDelayScript' value
+    }
+    deriving (Show)
+
+data Commands = JoinNetwork DiffTime
+               | Kill DiffTime
+               | Reconfigure DiffTime
+                             [(Int, Map RelayAccessPoint PeerAdvertise)]
```

Comment:
```suggestion
data Commands = JoinNetwork DiffTime
              | Kill DiffTime
              | Reconfigure DiffTime
                            [(Int, Map RelayAccessPoint PeerAdvertise)]
```

Reply:
I would rename the type to `Command`; then `[Command]` looks like a list of commands, rather than `[Commands]` (a list of list of commands).

Reply:
Addressed